16_Sockets_PR_for_review/homework_3_maximum/tcp_client_max.py

```diff
@@ -0,0 +1,155 @@
+"""
+# Homework, задача “хорошо”
+
+2. Реализовать TCP-client (или UDP-client):
+– Раз в минуту отправляет данные на сервер, данные на выбор:
+ • Данные CPU, температуры
+ • Количество секунд использования мышки
+ • Количество нажатий hot keys (ctrl+c, code inspect, ...)
+ • Свой вариант
+ – Формат данных: текущее время, название метрики, значение
+"""
+
+import json
+import psutil
+import logging
+import socket
+import time
+import win32api
+
+from datetime import datetime
+from typing import Dict, List, Any
+from threading import Thread
+
+
+CURSOR_X = 0
+CURSOR_Y = 0
+CLIENT_LOG_NAME = "client.log"
+TIME_INTERVAL = 60
+TIME_USE_MOUSE = 0
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = 9999
+
+
+def cpu_data() -> Dict[str, Any]:
+    """
+    System information about CPU
+    :return: Dictionary, where:
+    count_cpu - number of logical processors in the system
+    ctx_switches - number of context switches since boot
+    interrupts - number of interrupts since boot
+    soft_interrupts -number of software interrupts since boot
+    syscalls - number of system calls since boot
+    """
+    count_cpu = psutil.cpu_count()
+    stats_cpu = psutil.cpu_stats()
+    ctx_switches = stats_cpu.ctx_switches
+    interrupts = stats_cpu.interrupts
+    soft_interrupts = stats_cpu.soft_interrupts
+    syscalls = stats_cpu.syscalls
+
+    return {
+        "count_cpu": count_cpu,
+        "ctx_switches": ctx_switches,
+        "interrupts": interrupts,
+        "soft_interrupts": soft_interrupts,
+        "syscalls": syscalls,
+    }
+
+
+def get_cursor_position() -> List[int]:
+    """
+    Cursor position in screen coordinates
+    :return: Returns the cursor position in screen
+    coordinates - dot = (x, y)
+    """
+    return win32api.GetCursorPos()
+
+
+def time_use_mouse():
```

Comment:
Thank you for this implementation, in case of supporting several platforms (Linux, windows) we can use some packages for handling mouse event, one example: https://github.com/boppreh/mouse

Reply:
Thank you. An interesting project. Added to bookmarks.